src/test/resources/codeTagWithNestedBraces.txt

```diff
@@ -7,18 +7,17 @@
    |  |  |  |- TOKEN[type: CODE_LITERAL, text: code, line: 1, column: 51]
    |  |  |  |- TOKEN[type: TEXT, text:      Map<String, List<Map<String, Object>>> complexMap = new HashMap<>();, line: 2, column: 2]
    |  |  |  |- TOKEN[type: TEXT, text:      complexMap.put("key", Arrays.asList(, line: 3, column: 2]
-   |  |  |  |- TOKEN[type: TEXT, text:          Map.of("innerKey1", "{value1, line: 4, column: 2]
-   |  |  |  |- TOKEN[type: TEXT, text: }", "innerKey2", new Object() {, line: 4, column: 39]
-   |  |  |  |- TOKEN[type: TEXT, text:              @Override, line: 5, column: 2]
-   |  |  |  |- TOKEN[type: TEXT, text:              public String toString() {, line: 6, column: 2]
-   |  |  |  |- TOKEN[type: TEXT, text:                  return "Anonymous {Object} with @annotation and special chars like <, >, &";, line: 7, column: 2]
-   |  |  |  |- TOKEN[type: TEXT, text:              }, line: 8, column: 2]
-   |  |  |  |- TOKEN[type: TEXT, text:          }), line: 9, column: 2]
+   |  |  |  |- TOKEN[type: TEXT, text:          Map.of("innerKey1", ", line: 4, column: 2]
+   |  |  |  |- TOKEN[type: TEXT, text: {value1}, line: 4, column: 32]
+   |  |  |  |- TOKEN[type: TEXT, text: ", "innerKey2", new Object() , line: 4, column: 40]
+   |  |  |  |- TOKEN[type: TEXT, text: {\n *             @Override\n *             public String toString() {\n *                 return "Anonymous {Object} with @annotation and special chars like <, >, &";\n *             }\n *         }, line: 4, column: 69]
+   |  |  |  |- TOKEN[type: TEXT, text: ), line: 9, column: 12]
```

Comment:
This also helps us to eliminate a bunch of extra text nodes, which I wanted to do regardless. We can further condense our text nodes by using `more`.